skimage/io/tests/test_io.py

```diff
@@ -53,8 +44,10 @@ def test_imread_file_url():
     assert image.shape == (512, 512)
 
 
-def test_imread_http_url(urllib_urlopen):
-    query_string = '?' + 's' * 266
-    image_url = 'https://image.com/camera.jpg{}'.format(query_string)
-    image = io.imread(image_url)
+def test_imread_http_url(httpserver):
+    # httpserver is a fixture provided by pytest-localserver
```

Comment:
Does pytest need to be told that this exists, or is it picked up automatically?

Reply:
i think it is picked up automatically if you install `pytest-localserver`

Reply:
While testing, I was making it raise errors with print statements that I needed to better understand what was going on:

```
httpserver = <ContentServer(<class 'pytest_localserver.http.ContentServer'>, started 140405368153856)>

    def test_httpserver(httpserver):
        httpserver.serve_content(one_by_one_jpeg())
        # image = io.imread(httpserver.url + '?' + 's' * 266)
        image = io.imread(httpserver.url + '/test.jpg')
>       raise RuntimeError(httpserver.url)
E       RuntimeError: http://127.0.0.1:44591

```
seems to create a server on localhost, and an empty port.

The it probably just returns whatever binary data you give it.

Reply:
Yes, that's perfect.